keyboards/soldered/macro_pad/keyboard.json

```diff
@@ -0,0 +1,41 @@
+{
+    "manufacturer": "Soldered Electronics",
+    "keyboard_name": "Soldered Macro Pad",
+    "maintainer": "SolderedElectronics",
+    "bootloader": "rp2040",
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "GP18", "pin_b": "GP20", "resolution": 4}
+        ]
+    },
+    "features": {
+        "bootmagic": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["GP9", "GP11", "GP13", "GP10", "GP12", "GP19"],
+        "rows": ["GP0"]
```

Comment:
More of an FYI, direct pin matrix would have been better.  No row pin would have been needed, and each pin could have been wired directly to GPIO, and no diodes would have been needed.  Also, direct pin is MUCH faster scanning (which isn't as important here).